docs/proposals/respect-rbac-for-resource-exclusions.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Respect RBAC for Resource Inclusions/Exclusions
+
+authors:
+- "@gdsoumya"
+- "@alexmt"
+
+sponsors:
+- TBD
+
+reviewers:
+- TBD
+
+approvers:
+- TBD
+
+creation-date: 2023-05-03
+
+---
+
+# Enhancement Idea
+
+This is a proposal to provide the ability to configure argocd controller, to respect the current RBAC permissions 
+when handling resources besides the already existing resource inclusions and exclusions.
+
+## Summary
+
+Argo CD administrator will be able to configure in `argocd-cm`, whether to enable or disable(default) the feature where the controller will 
+only monitor resources that the current service account allows it to read.
+
+## Motivation
+
+Some users restrict the access of the argocd to specific resources using rbac and this feature will enable them to continue 
+using argocd without having to manually configure resource exclusions for all the resources that they don't want argocd to be managing.
+
+## Proposal 
+
+The configuration for this will be present in the `argocd-cm`, we will add new boolean field `resource.respectRBAC` in the
+cm which can be set to `true` to enable this feature, by default the feature is disabled.
```

Comment:
Wouldn't this be safe to enable by default? If the controller doesn't have access to a resource, respecting RBAC will only help it avoid 403s.

Reply:
We can keep it enabled by default, but that might suddenly change the behavior of the controller when a user upgrades which I thought might not be welcomed by some users.

Reply:
Makes sense! I think we'd want to everywhere target enabling by default. But agreed,  best to move cautiously.

Reply:
I would also be voting for kind of a cautious move here. Make it a feature toggle, test it extensively, and then move forward to enable it by default.